### ISSUE TITLE
Support AMP consentType, gdprApplies, additionalConsent

### ISSUE THREAD

**Issue by bretg** [prebid/prebid-server#1817]
We're a bit behind on this, but as noted in AMP issue 30006, Prebid Server can pull several new values from the AMP interface as documented in https://github.com/ampproject/amphtml/blob/main/extensions/amp-consent/integrating-consent.md#consentmetadata

1. consent_type=CONSENT_METADATA(consentStringType)
1. gdpr_applies=CONSENT_METADATA(gdprApplies)
1. addtl_consent=CONSENT_METADATA(additionalConsent)

e.g.

```
...&consent_type=2&gdpr_applies=true&addtl_consent=aaaaaaaaaaaaaa&...
```

## Details

Note that the AMP parameter gdpr_consent carries both GDPR and CCPA values.

1. If consent_type is provided, it will be an enum containing the following values:
   - 1: The contents of gdpr_consent can be treated as TCF V1. We no longer support TCFv1, so ignore any consent_string provided on the request.
   - 2: The contents of gdpr_consent can be treated as TCF V2. If the consent_string isn't a valid TCF2 string, assume there's no user consent for any purpose as if no gdpr_consent were provided.
   - 3: The contents of gdpr_consent can be treated as a US Privacy String. If the consent_string isn't a valid USP string, assume for the purposes of opt-out-of-sale enforcement that no gdpr_consent was provided.
   - anything else: ignore, log a warning, and assume no gdpr_consent was provided
1. If gdpr_applies is supplied, use its value to set `regs.ext.gdpr`: gdpr_applies=true means regs.ext.gdpr:1, gdpr_applies=false means regs.ext.gdpr:0, any other value means regs.ext.gdpr is not set.
1. If consent_type="2", and gdpr_consent is not empty, then copy gdpr_consent to user.ext.consent
1. If consent_type="3", and gdpr_consent is not empty, then copy gdpr_consent to regs.ext.us_privacy
1. If addtl_consent is supplied, copy its value to `user.ext.ConsentedProvidersSettings.consented_providers`

**Comment by snahornyi** [comment]
@bretg You mentioned that `gdpr_applies=1` this param can be of number value, but from doc we can see that it can be only boolean param(`true` or `false`), so should `0` and `1` also be supported?

**Comment by bretg** [comment]
example fixed - thanks for pointing it out

**Comment by bretg** [comment]
done in PBS-Java with https://github.com/prebid/prebid-server-java/pull/1379

**Comment by bretg** [comment]
FYI - there was an inconsistency between the original version of this issue and what was on the AMP PR. Changed the parameter from "attl_consent" to "addtl_consent".

**Comment by bretg** [comment]
Released in PBS-Java 1.70

**Comment by mirkorean** [comment]
Any update on this issue for PBS-Go? As our adapter is currently expecting both the gdpr flag and the consent string, we are trying to see if we need to change this there or if we can expect an implementation of the gdpr_applies parameter in AMP.

Not content enough with the code base yet to try a PR ourselves for this issue. Thanks!

**Comment by guscarreon** [comment]
> `5. If addtl_consent is supplied, copy its value to user.ext.ConsentedProvidersSettings.consented_providers`

PBS-Go will parse the `addtl_consent` string into an array of integers located at `user.ext.consented_providers_settings.consented_providers`:
```
  "user": {
    "ext": {
      "consented_providers_settings": {
        "consented_providers": [
          10,
          20,
          90
        ]
      }
    }
  },
```

**Comment by bretg** [comment]
Thanks. Will open a ticket to change to user.ext.consented_providers_settings.consented_providers in Java.

**Comment by SyntaxNode** [comment]
Implemented in PBS-Go 0.231.0.
